<?xml version="1.0"?>
<launch>
    <arg name="master_name"         default="$(env MASTER)"/>
    <arg name="robot_name"          default="$(env HOST)"/>
    <arg name="enable_navigation"   default="true"/>

    <!--topic和frame设置-->
    <arg      if="$(eval robot_name == '/')"    name="topic_prefix" default=""/>
    <arg unless="$(eval robot_name == '/')"     name="topic_prefix" default="/$(arg robot_name)"/>
    <arg      if="$(eval robot_name == '/')"    name="frame_prefix" default=""/>
    <arg unless="$(eval robot_name == '/')"     name="frame_prefix" default="$(arg robot_name)/"/>

    <arg     if="$(eval robot_name == master_name)" name="map_frame" default="$(arg frame_prefix)map"/>
    <arg unless="$(eval robot_name == master_name)" name="map_frame" default="$(arg master_name)/map"/>
    <arg name="clicked_point"       default="$(arg topic_prefix)/clicked_point"/>
    <arg name="move_base_result"    default="$(arg topic_prefix)/move_base/result"/>

    <group ns="$(arg robot_name)">
        <!--多点导航-->
        <node pkg="jetauto_navigation" type="publish_point.py" name="publish_point" output="screen">
            <param name="map_frame"         value="$(arg map_frame)"/>
            <param name="clicked_point"     value="$(arg clicked_point)"/>
            <param name="move_base_result"  value="$(arg move_base_result)"/>
            <param name="enable_navigation" value="$(arg enable_navigation)"/>
        </node>
    </group>
</launch>
